lib/matplotlib/cbook.py

```diff
@@ -1661,7 +1663,10 @@ def safe_isfinite(val):
         raise RuntimeError("matplotlib does not "
                            "support generators as input")
     else:
-        return next((val for val in obj if safe_isfinite(val)), safe_first_element(obj))
+        for val in obj:
+            if safe_isfinite(val):
+                return val
+        return safe_first_element(obj)
```

Comment:
Is this really faster? I have no idea, but was under the impression that the generator would stop at the first element? May for sure be wrong though.

Reply:
This form will avoid calling `safe_first_element` unless we actually need it where as in the `next` from we always evaluate it.